core/modules/layout/layout.module

```diff
@@ -90,6 +92,105 @@ function layout_menu() {
     'weight' => 20,
   ) + $base;
 
+  // Put all code in separate file to make dev easier, because layout.admin is huge.
+  // todo change path to flexible-template
+  $items['admin/structure/layouts/settings/flexible-template/add'] = array(
+    'title' => 'Add flexible layout template',
+    'page callback' => 'backdrop_get_form',
+    'page arguments' => array('layout_flexible_template_settings_form'),
+    'type' => MENU_LOCAL_ACTION,
+    'weight' => -10,
+    'access arguments' => array('administer layouts'),
+    'file' => 'layout.flexible.inc',
+  );
+
+  $items['admin/structure/layouts/settings/flexible-template/%/edit'] = array(
```

Comment:
In all cases where a flexible layout template is being loaded, using a magic menu loader might be better in each menu path.
```suggestion
  $items['admin/structure/layouts/settings/flexible-template/%layout_flexible_template/edit'] = array(
```

That will automatically load the flexible template into the menu callback instead of getting the `$flexible_template_id`, it will be the `$flexible_template` object. It will also automatically make any paths that are not a valid template ID return a 404 not found if the ID does not exist.

Reply:
This was the plan all along. Done.